### PR TITLE
Improve readability of _mathtext_data.stix_virtual_fonts table

### LINKED ISSUE #25738: [MNT]: Improve readability of _mathtext_data.stix_virtual_fonts table
### Summary

Reviewing https://github.com/matplotlib/matplotlib/pull/25359/files#diff-49ee499471e3397e95aa265c6694a643c8c16c625c700da06b9c3c58db69fc52 I noted that an annoying part is just looking at the codepoint values.  One idea to improve readability would be to replace lines such as
```python
(0x0393, 0x03a9, 'bfit', 0x1d71E), # \Gamma-\Omega
```
by
```python
("\N{GREEK CAPITAL LETTER GAMMA}", "\N{GREEK CAPITAL LETTER OMEGA}", "bfit",
 "\N{MATHEMATICAL BOLD ITALIC CAPITAL GAMMA}"),
```
which seems much more self-documenting (and adapting the table reading code to support that seems easy enough and have negligible cost -- just a matter of calling ord() in a few places).

### Proposed fix

_No response_

### PR DESCRIPTION
**Pull request by lvr44**
## PR summary
Hi! I made some changes on mathtext_data file to improve readability (using Unicode character name in order to make to code clearer) as asked (issue 25738).

So, i changed the table from like this:
```
(0x0393, 0x0393, 'it', 0x213e),  # \Gamma (not in beta STIX fonts)
(0x03a0, 0x03a0, 'it', 0x213f),  # \Pi
```

To this:
```
 ("\N{GREEK CAPITAL LETTER GAMMA}", "\N{GREEK CAPITAL LETTER GAMMA}",
     'it', 0x213e),
 ("\N{GREEK CAPITAL LETTER PI}", "\N{GREEK CAPITAL LETTER PI}",
     'it', 0x213f),
```
## PR checklist

- [x] "closes #25738" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/_mathtext_data.py [hunk 1]
```diff
@@ -1056,11 +1056,16 @@
             (0x0053, 0x0059, 'rm', 0x1d54a), # S-Y
             (0x005a, 0x005a, 'rm', 0x2124),  # Z
             (0x0061, 0x007a, 'rm', 0x1d552), # a-z
-            (0x0393, 0x0393, 'rm', 0x213e),  # \Gamma
-            (0x03a0, 0x03a0, 'rm', 0x213f),  # \Pi
-            (0x03a3, 0x03a3, 'rm', 0x2140),  # \Sigma
-            (0x03b3, 0x03b3, 'rm', 0x213d),  # \gamma
-            (0x03c0, 0x03c0, 'rm', 0x213c),  # \pi
+            (ord("\N{GREEK CAPITAL LETTER GAMMA}"),
```
Comment (anntzer) [inline lib/matplotlib/_mathtext_data.py:1059]: I would also replace the codepoints above (0x0030 = `"0"` (no real need to write that as `"\N{DIGIT ZERO}"`)), as well as the last items of the tuple *when applicable* (0x213e = `"\N{DOUBLE-STRUCK CAPITAL GAMMA}"` but 0x1d7d8 is just an internal codepoint for the font with no associated name).

Comment (lvr44) [reply lib/matplotlib/_mathtext_data.py:1072]: Hi, thanks for the feedback! I included replacements for these codepoints and for the last items that i could find as well in my last commit

**Comment by QuLogic** [comment]
As noted in the issue, I don't think you want to write `ord` everywhere, and instead replace that with a dict/list comprehension that calls it in one place for everything.

**Comment by lvr44** [comment]
Thanks for the feedback @QuLogic! I've updated the code to include the list and dict comprehension now. For the unicodes list i had to include "\N{}" in the strings so it wouldn't throw an error when it loops because of "0", "9" and the letters strings. Btw, i really couldn't identify why there is a linting problem

**Review by QuLogic — COMMENTED** [review]
I don't think there's a need for `list_uni` or `uni_map`; that's basically a longer way of writing `ord`. Rather, just write out the text in each of the tables, and do one dict/list comprehension at the end over everything in `stix_virtual_fonts`. If you need to mix characters and ordinals, then perhaps only do that on the lists that need it directly.

**Pull request closed without merging**